mne_connectivity/spectral/epochs.py

```diff
@@ -1123,39 +1871,60 @@ def spectral_connectivity_epochs(data, names=None, method='coh', indices=None,
 
     # compute final connectivity scores
     con = list()
-    for conn_method, n_args in zip(con_methods, n_comp_args):
+    patterns = list()
+    for method_i, conn_method in enumerate(con_methods):
+
         # future estimators will need to be handled here
-        if n_args == 3:
-            # compute all scores at once
-            conn_method.compute_con(slice(0, n_cons), n_epochs)
-        elif n_args == 5:
+        if conn_method.accumulate_psd:
             # compute scores block-wise to save memory
             for i in range(0, n_cons, block_size):
                 con_idx = slice(i, i + block_size)
                 psd_xx = psd[idx_map[0][con_idx]]
                 psd_yy = psd[idx_map[1][con_idx]]
                 conn_method.compute_con(con_idx, n_epochs, psd_xx, psd_yy)
         else:
-            raise RuntimeError('This should never happen.')
+            # compute all scores at once
+            if method[method_i] in _multivariate_methods:
+                conn_method.compute_con(indices_use, rank, n_epochs)
+            else:
+                conn_method.compute_con(slice(0, n_cons), n_epochs)
 
         # get the connectivity scores
         this_con = conn_method.con_scores
+        this_patterns = conn_method.patterns
 
-        if this_con.shape[0] != n_cons:
-            raise ValueError('First dimension of connectivity scores must be '
-                             'the same as the number of connections')
+        assert this_con.shape[0] == n_cons, (
+            'first dimension of connectivity scores does not match the '
+            'number of connections; please contact the mne-connectivity '
+            'developers')
```

Comment:
We generally don't want `assert` messages as these do not get run if the Python package binary is distributed.

Reply:
**FIXED**